examples/threading/threading.c

```diff
@@ -0,0 +1,105 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+// Optional: use these functions to add debug or error prints to your application
+//#define DEBUG_LOG(msg,...)
+#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+#define SUCCESS   (0)
+#define FAILURE   (-1)
+#define CONVERT_MS_TO_US(value)  ((1000)*value)
+
+void* threadfunc(void* thread_param)
+{
+    int return_value = FAILURE;
+    
+    if (NULL != thread_param)
+    {
+        // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+        // hint: use a cast like the one below to obtain thread arguments from your parameter
+        struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+        usleep(CONVERT_MS_TO_US(thread_func_args->wait_to_obtain_ms));
```

Comment:
@chchbuff is it possible for `usleep` to fail to sleep for the specified amount of time?